src/hooks/useMultiSelect.ts

```diff
@@ -0,0 +1,113 @@
+import { useState } from "react";
+import {
+  MultiSelectItem,
+  SelectedItems,
+} from "../components/MultiSelect/MultiSelect";
+
+interface useMultiSelectProps {
+  multiSelectId: string;
+  items: MultiSelectItem[];
+}
+
+/**
+ * The useMultiSelect The hook returns an object containing all the props and state needed to handle the selectedItems.
+ * That includes the functions onChange, onClear, onMixedStateChange for handling any changes to the selection of items and the current state of the selection: selectedItems.
+ */
+export default function useMultiSelect({
+  multiSelectId,
+  items,
+}: useMultiSelectProps) {
+  const [selectedItems, setSelectedItems] = useState<SelectedItems | {}>({});
+
+  function handleChange(itemId: string) {
+    let itemIds;
+    // Check if the id already exists in the state.
+    if (selectedItems[multiSelectId] !== undefined) {
+      let itemIdExists =
+        selectedItems[multiSelectId].items.indexOf(itemId) > -1;
+      // Make a copy of the existing items array.
+      itemIds = selectedItems[multiSelectId].items.slice();
+      // If id exists, remove it from the array.
+      if (itemIdExists) {
+        itemIds = itemIds.filter((id) => id !== itemId);
+      } else {
+        // Add it to the array, but modify the copy, not the original.
+        itemIds.push(itemId);
+      }
+    } else {
+      itemIds = [];
+      itemIds.push(itemId);
+    }
+
+    setSelectedItems({
+      ...selectedItems,
+      [multiSelectId]: {
+        items: itemIds,
+      },
+    });
+  }
+
+  /**
+   * hanldeMixedStateChange is only used for the "dailog" variant. It hanldes the state for checkbox options with child options.
```

Comment:
Some typos: "hanlde" should be "handle", "dailog" should be "dialog", "hanldes" should be "handles", and a missing "t" after "accep".

Reply:
fixed